src/main/java/com/dpgil/pathlinker/path_linker/internal/Algorithms.java

```diff
@@ -759,8 +764,70 @@ public static CyEdge getEdge(
             }
         }
 
+        // If undirected edges are allowed, then also check for an undirected edge between them
+        // Currently directed edges are favored over undirected edges
+        if (includeUndirected) 
+        {
+			List<CyEdge> undirConnections =
+				network.getConnectingEdgeList(source, target, CyEdge.Type.UNDIRECTED);
+			for (CyEdge edge : undirConnections)
+			{
+				return edge;
+			}
+        	
+        }
+
         return null;
     }
+    
+    
+    /**
+     * Returns 
+     * @param network
+     * 				the network
+     * @param node
+     * 				the node to get the connecting edges
+     * @param edgeType
+     * 				Should be either CyEdge.Type.OUTGOING or CyEdge.Type.INCOMING
+     * @return List<CyEdge> the list of edges (either outgoing or incoming) connected to the given node
+     */
+    public static List<CyEdge> getAdjacentEdgeList(CyNetwork network, CyNode node, CyEdge.Type edgeType)
+    {
+		
+    	// retrieves all of the directed edges incoming to or outgoing from the given node
+    	List<CyEdge> adjacentEdges = network.getAdjacentEdgeList(node, edgeType);
+
+    	// If specified, add all of the undirected edges connected to the given node as well
+    	if (includeUndirected){
```

Comment:
I originally passed the variable includeUndirected in as a parameter to each of the three functions getAdjacentEdgeList(), getNeighborList() and getEdge(), but I couldn't think of a reason a user would want to include undirected edges only part of the time (i.e. for one of those functions but not for the other). I thought it would be cleaner to just have one variable they all use. I also didn't think those functions would be used to retrieve only undirected edges so I didn't include that case